solr/core/src/java/org/apache/solr/handler/RequestHandlerBase.java

```diff
@@ -319,6 +325,11 @@ public PluginInfo getPluginInfo(){
     return  pluginInfo;
   }
 
+  @Override
+  public void close()  {
+    if (metricsInfo != null) metricsInfo.unregister();
```

Comment:
Again, this is also a repeating pattern that is easy to forget (in fact you forgot about this in one place even in this patch). If it's something that has to be done reliably then I think the framework should take care of that.

See for example how `SolrInfoBean.registerMetricName()` is designed - it uses result of another method to determine whether it's a no-op. Similarly, we could add `SolrMetricProducer.getMetricsInfo()` in addition to `AutoCloseable` and then always call `unregister` in `default void close()`.